tests/ops/op_math/test_composite.py

```diff
@@ -258,3 +258,33 @@ def test_depth_property(self):
 
         op = ValidOp(qml.PauliX(0), ValidOp(qml.Identity(wires=0), qml.RX(1.9, wires=1)))
         assert op.arithmetic_depth == 2
+
+    def test_overlapping_ops_property(self):
+        """Test the overlapping_ops property."""
```

Comment:
Also worth adding a test which checks if the cached private property gets update as expected after repeated calls?

Reply:
Done!